saleor/account/migrations/0083_confirm_active_user.py

```diff
@@ -0,0 +1,52 @@
+# Generated by Django 3.2.24 on 2024-02-29 13:21
+
+from django.db import migrations, transaction
+from django.db.models import QuerySet
+
+# Batch size of 5000 is about ~8MB of memory usage
+BATCH_SIZE = 5000
+
+
+def queryset_in_batches(queryset):
+    """Slice a queryset into batches.
+
+    Input queryset should be sorted be pk.
+    """
+    start_pk = 0
+
+    while True:
+        qs = queryset.filter(pk__gt=start_pk)[:BATCH_SIZE]
+        pks = list(qs.values_list("pk", flat=True))
+
+        if not pks:
+            break
+
+        yield pks
+
+        start_pk = pks[-1]
+
+
+def set_user_is_confirmed_to_true(qs: QuerySet):
+    with transaction.atomic():
+        # lock the batch of objects
+        _users = list(qs.select_for_update(of=(["self"])))
```

Comment:
I don't think that you need to lock the DB, when executing the code in migration process 🤔